File: src/main/java/io/vertigo/shiny/components/media/geomap/ShinyGeoMap.java
package io.vertigo.shiny.components.media.geomap;

import java.util.List;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinyGeoMap(
		String title,
		List<ShinyGeoPoint> geoPoints) implements ShinyComponent {

	@Override
	public String type() {
		return "geoMap";
	}


	public ShinyGeoMap {
		Assertion.check()
				.isNotBlank(title, "Title cannot be blank");
		Assertion.check()
				.isNotNull(geoPoints, "GeoPoints list cannot be null");
	}
}

File: banshee/src/models/ShinyGeoMap.ts
import { ShinyComponent } from '../ShinyComponent';
import { ShinyGeoPoint } from './ShinyGeoPoint';

export class ShinyGeoMap implements ShinyComponent {
  title: string;
  geoPoints: ShinyGeoPoint[];
  type: string = 'geoMap';

  constructor(title: string, geoPoints: ShinyGeoPoint[]) {
    this.title = title;
    this.geoPoints = geoPoints;
  }
}
